src/env_var.rs

```diff
@@ -45,3 +47,52 @@ pub fn inc(name: &str, cmd: &mut Command) {
 
     cmd.env(name, (old_value + 1).to_string());
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::currentprocess;
+    use crate::test::{with_saved_path, Env};
+
+    use std::collections::HashMap;
+    use std::ffi::{OsStr, OsString};
+
+    #[test]
+    fn prepent_unique_path() {
+        let mut vars = HashMap::new();
+        vars.env("PATH", "/home/a/.cargo/bin:/home/b/.cargo/bin");
+        let tp = Box::new(currentprocess::TestProcess {
+            vars,
+            ..Default::default()
+        });
+        with_saved_path(&|| {
+            currentprocess::with(tp.clone(), || {
+                let mut path_entries = vec![];
+                let mut cmd = Command::new("test");
+
+                // duplicated PATH.
+                let a = OsString::from("/home/a/.cargo/bin");
+                let path_a = PathBuf::from(a);
+                path_entries.push(path_a);
+                // unique PATH.
+                let z = OsString::from("/home/z/.cargo/bin");
+                let path_z = PathBuf::from(z);
+                path_entries.push(path_z);
+
+                prepend_path("PATH", path_entries, &mut cmd);
+
+                let envs: Vec<(&OsStr, Option<&OsStr>)> = cmd.get_envs().collect();
```

Comment:
one way you could avoid using this unstable method is to have the code you're testing depend on a trait defined in this module.

Implement it once for Command, and once for a trivial test struct (or use mockall), and then you can assert on what the set env is, rather than interacting with Command at all.

Reply:
I'm trying to reimplement the unit test but I'm stuck on this part. Which trait should I depend on? 
> a trait defined in this module.

Reply:
```
pub fn append_path(
    name: &str,
    value: Vec<PathBuf>,
    cmd: &mut dyn ProcessEnvs<Item = Command, Iter = vec::IntoIter<Command>>,
) {
```
I am stack on `the trait `ProcessEnvs` cannot be made into an object` error. I tried to separate the trait but it didn't work. This is the latest [commit](https://github.com/rust-lang/rustup/pull/2849/commits/afe83e0c0fc194c3be4e492309dd38b5a1b95c99).